src/transform/src/equivalence_propagation.rs

```diff
@@ -250,9 +252,12 @@ impl EquivalencePropagation {
                     .value::<Equivalences>()
                     .expect("Equivalences required");
 
-                if let Some(input_equivalences) = input_equivalences {
-                    // Clone the equivalences in case of variadic map, which will need to mutate them.
-                    let mut input_equivalences = input_equivalences.clone();
+                if let Some(input_equivalences_orig) = input_equivalences {
+                    let mut input_equivalences_cloned = None;
+                    if !ctx.features.enable_dequadratic_eqprop_map {
+                        // We mutate them for variadic Maps if the feature flag is not set.
+                        input_equivalences_cloned = Some(input_equivalences_orig.clone());
+                    }
```

Comment:
It's unclear from the context and the comments why this optional cloning exists. Given the complexity the PR adds, we should also add the corresponding explanations to know how to dig ourselves back out in the future.

Reply:
Ok, I've added a comment.